### PR TITLE
[Doc] Link to doc/contributing.rdoc from CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by kachick**
https://github.com/ruby/ruby/blob/c7e6914b3947cdf0e9c0d28d1162a084d0138887/doc/contributing.rdoc looks the active one of `How to contribute`. So adding link in `CONTRIBUTING.md` might help someone like me 🙏

In my case...
I often forgot `How to test specific test file with the built ruby?`
Nothing in README. And I can't find it in https://bugs.ruby-lang.org/projects/ruby/wiki/HowToContribute

But the rdoc describes as below.

```
This is also how you can run a specific test from our build dir:

make test-all TESTS=drb/test_drb.rb
```

It solved all. ☺️

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,4 +1,5 @@
-Please see the [official issue tracker] and wiki [HowToContribute].
+Please see the [official issue tracker], [contributing.rdoc] and wiki [HowToContribute].
 
 [official issue tracker]: https://bugs.ruby-lang.org
+[contributing.rdoc]: https://github.com/ruby/ruby/blob/master/doc/contributing.rdoc
```
Comment (nobu) [inline CONTRIBUTING.md:4]: Shouldn't it be a link to the local file?

```suggestion
[contributing.rdoc]: rdoc-ref:contributing.rdoc
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,4 +1,5 @@
-Please see the [official issue tracker] and wiki [HowToContribute].
+Please see the [official issue tracker], [contributing.rdoc] and wiki [HowToContribute].
```
Comment (nobu) [inline CONTRIBUTING.md:1]: To be valid when reading as a plain text, it would need to be prefixed by `doc/`.

File: CONTRIBUTING.md (hunk 1 above)
Comment (kachick) [reply CONTRIBUTING.md:4]: Oh, I don't know the feature `rdoc-ref`! 😮
Is it a `RDoc` feature? So can't we use it in Markdown?

https://github.com/ruby/ruby/blob/db0ad48309edae28a65e3d18e9b3a15753eda777/doc/NEWS-3.0.0.md

Above doc having similar style, it shows GFM as below

スクリーンショット 2021-03-22 1 10 15

And this change so on...

https://github.com/ruby/ruby/commit/c5795c37e61de30df15501e48ad61167e5841953
https://github.com/ruby/ruby/blob/c5795c37e61de30df15501e48ad61167e5841953/CONTRIBUTING.md

スクリーンショット 2021-03-22 1 10 35

Do we need to handle `.md` files as a rdoc in this repository?

File: CONTRIBUTING.md (hunk 2 above)
Comment (kachick) [reply CONTRIBUTING.md]: Thank you for the suggestion! Is https://github.com/ruby/ruby/pull/4266/commits/0387454fc0e18c75df86eefa390d0d4ced403205 correct understanding?

File: CONTRIBUTING.md (hunk 1 above)
Comment (nobu) [reply CONTRIBUTING.md]: It is possible in also RDoc's Markdown, but GitHub doesn't support it of course, even in *.rdoc files.
In RDoc, "rdoc-ref:" link generates a URL from the path under the certain rule.
I made ruby/rdoc#800 to enable it for explicit references without the prefix.
Once this gets applied, the above link no longer needs the prefix.

Comment (kachick) [reply CONTRIBUTING.md]: Thank you for the rdoc improvement! 😂 

I have applied your suggestion https://github.com/ruby/ruby/pull/4266/commits/cc779f4e70fac9ffb69925571449a461c918ca92

```console
$ rdoc --version
6.3.0
$ rdoc CONTRIBUTING.md --op tmp-rdoc
$ open tmp-rdoc/CONTRIBUTING_md.html
```

スクリーンショット 2021-03-22 13 53 20

It looks not a link 🤔 
It makes same even if adding `doc/` prefix as `rdoc-ref:doc/contributing.rdoc`.
But https://docs.ruby-lang.org/en/master/doc/NEWS-3_0_0_md.html shows as link.

スクリーンショット 2021-03-22 13 55 19

Hmm… Definitely I do not know something about rdoc :bow:

Comment (nobu) [reply CONTRIBUTING.md]: ruby/rdoc#800 is not merged yet.
And it doesn't change the result for "rdoc-ref:", the line needs to be `[doc/contributing.rdoc]: doc/contributing.rdoc` without the schema prefix.

Comment (kachick) [reply CONTRIBUTING.md]: https://github.com/ruby/ruby/blob/f2ad63b361236df26aae33a19c851f106e752b1b/CONTRIBUTING.md

スクリーンショット 2021-03-22 17 32 46

Ah… I may understand it finally. 😓 
https://github.com/ruby/ruby/pull/4266/commits/f2ad63b361236df26aae33a19c851f106e752b1b is this correct understanding?
I didn't know Markdown can use relative path, This is so useful!

Comment (nobu) [reply CONTRIBUTING.md]: Exactly.

**Comment by kachick** [comment]
Thank you! 🙏 

Of course I am appreciating to https://github.com/ruby/rdoc/pull/800 😂

**Pull request merged**